xkafka/consumer.go

```diff
@@ -0,0 +1,162 @@
+package xkafka
+
+import (
+	"context"
+	"errors"
+	"strings"
+	"time"
+
+	"github.com/confluentinc/confluent-kafka-go/kafka"
+	"github.com/sourcegraph/conc/stream"
+
+	"github.com/gojekfarm/xtools/xkafka/internal"
+)
+
+// ConsumerFunc allows the user to customise confluent-kafka-go/kafka.Consumer.
+type ConsumerFunc func(cfg *kafka.ConfigMap) (internal.ConsumerClient, error)
```

Comment:
It is usually a bad idea to expose dependent library in the public API, we can either map this to new type(we already have `kafka.ConfigMap` ~~ `xkafka.ConfigMap`), or we should create type aliases.